pkg/bpf/map_linux.go

```diff
@@ -124,11 +127,13 @@ type Map struct {
 }
 
 // NewMap creates a new Map instance - object representing a BPF map
-func NewMap(name string, mapType MapType, keySize int, valueSize int, maxEntries int, flags uint32, innerID uint32, dumpParser DumpParser) *Map {
+func NewMap(name string, mapType MapType, mapKey MapKey, keySize int, mapValue MapValue, valueSize int, maxEntries int, flags uint32, innerID uint32, dumpParser DumpParser) *Map {
```

Comment:
I don't think it's a good idea because of the issues that we had in the past? I.e. the differences between size that go was aware for go structures and C structures.

Reply:
I think it's fine to derive the size using the `reflect` library, as the issue we had was with the CGO generated code.